service/gateway/metadata_handler.go

```diff
@@ -96,14 +111,75 @@ func (gateway *Gateway) listObjectsByBucketNameHandler(w http.ResponseWriter, r
 		return
 	}
 
-	if err = s3util.CheckValidBucketName(reqContext.bucketName); err != nil {
-		log.Errorw("failed to check bucket name", "bucket_name", reqContext.bucketName, "error", err)
+	queryParams = reqContext.request.URL.Query()
+	requestBucketName = reqContext.bucketName
+	requestMaxKeys = queryParams.Get("max_keys")
+	requestStartAfter = queryParams.Get("start_after")
+	requestContinuationToken = queryParams.Get("continuation_token")
+	requestDelimiter = queryParams.Get("delimiter")
+	requestPrefix = queryParams.Get("prefix")
```

Comment:
The constant queries field related to restful api can be defined in model const.go

Reply:
fixed